cmd/object-api-datatypes.go

```diff
@@ -415,6 +417,20 @@ type ListObjectsV2Info struct {
 	Prefixes []string
 }
 
+func concurrentDecryptETag(ctx context.Context, objects []ObjectInfo) {
+	g := errgroup.WithNErrs(len(objects))
+	for index := range objects {
+		index := index
+		g.Go(func() error {
+			objects[index].ETag = objects[index].GetActualETag(nil)
+			objects[index].Size, _ = objects[index].GetActualSize()
+			return nil
+		}, index)
+	}
+	g.Wait()
+	return
+}
```

Comment:
Same, but simplifies the caller.
```suggestion
func concurrentDecryptETag(ctx context.Context, objects []ObjectInfo) {
	const maxConcurrent = 500
	for len(objects) > 0 {
		todo := objects
		if len(todo) > maxConcurrent {
			todo = todo[:maxConcurrent]
		}
		objects = objects[len(todo):]
		g := errgroup.WithNErrs(len(todo))
		for index := range todo {
			index := index
			g.Go(func() error {
				todo[index].ETag = objects[index].GetActualETag(nil)
				todo[index].Size, _ = objects[index].GetActualSize()
				return nil
			}, index)
		}
		g.Wait()
	}
	return
}
```